card_lib/ranks.py

```diff
@@ -0,0 +1,87 @@
+"""
+Classes to rank cards based on various card rules.
+"""
+from abc import ABC, abstractmethod
+
+
+class CardRanksInterface(ABC):
+    """
+    Abstract interface to access rules around card rankings.
+    """
+    @abstractmethod
+    def pip_score(self, card):
+        """
+        Score of the pip, used when calculating points at
+        the end of a game. Depending on the rules of a game, aces
+        can be considered high or low.
+
+        Returns:
+            score (int)
+        """
+        pass
+
+    @abstractmethod
+    def pip_ordinal(self, card):
+        """
+        Ordinal number for a given pip, which can be used
+        in comparing with other pips, i.e. 10=10, J=11, etc...
+
+        Returns:
+            ordinal (int)
+        """
+        pass
+
+    def beats(self, card1, card2):
```

Comment:
I am imagining the use case for 'beats', and it seems like once you call it, you would then need to check which card was returned (which deck it came from) in order to be able to make a decision. This seems like extra work. Am I over-thinking this?

Reply:
I think you're correct. I'm going to modify the approach here.

Reply:
On second thought, aside from a clunky method name which I'll change, this impl seems right to me. The problem is that there's three possible conditions here:
- card1 wins
- card2 wins
- tie
Do you have other ideas about communicating the three statuses back aside from returning the winner or `None`?

Reply:
That seems like a reasonable way forward. In the case of a tie, there is no winner.